### PR TITLE
Hb3

### PR DESCRIPTION
**Pull request by jah701**

### CODE REVIEW

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            session.close();
+        }
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<User> query = session.createQuery("from User u where email = :email");
+            query.setParameter("email", email);
+            return Optional.of(query.getSingleResult());
```
Comment (matviichuk92) [inline src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java]:
```suggestion
            return query.uniqueResultOptional();
```

**Review by matviichuk92 — APPROVED** [review]
good job!

File: src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java [hunk 2]
```diff
@@ -1,4 +1,44 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(user);
+            transaction.commit();
+            return user;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add new user", e);
```
Comment (matviichuk92) [inline src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java]: Mentors recommended making this message more informative
```suggestion
            throw new DataProcessingException("Can't add new " + user, e);
```

File: src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFromDb = userService.findByEmail(email);
+        byte[] salt = userFromDb.get().getSalt();
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java]: тут може бути NPE

File: src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        String saltedPass = HashUtil.hashPassword(password, salt);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java]:
```suggestion
        String saltedPassword = HashUtil.hashPassword(password, salt);
```

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**